classes/class4/homework/routes/cats.js

```diff
@@ -0,0 +1,95 @@
+module.exports = function(){
+    var express = require('express');
+    var router = express();
+    var Cat = require('../models/cat_model.js')
+
+    router.get('/', function(req,res,next) {
+      Cat.find({}).sort({age: -1}).exec(function(err,cats){
+        if (err) {
+            console.log('Error displaying all cats!');
+            res.status(404).send('Error displaying all cats!');
+        } else {
+            cats = cats.map(function (cat){
+                return 'name: '+cat['name']+' age: '+cat['age']+' colors: '+cat['colors'];
+            });
+            res.render('home',{'cats': cats});
+        }});
+    });
+
+    router.get('/new', function(req,res,next) {
+        function getRandomInt() {
+          var min_age = 1;
+          var max_age = 100;
+          return Math.floor(Math.random() * (max_age - min_age)) + min_age;
+        }
+        function getRandomName() {
+          var names = ['evan','ben','sarah','josh','thomas','paul','allen','another evan'];
+          return names[Math.floor(Math.random()*names.length)];
+        }
+        function getRandomColors() {
+          var colors = ['blue','green','yellow','pink','purple','sunshine','cat-color'];
+          var random_colors = [];
+          for (var i=0; i < colors.length;i++){
+              if (Math.random() < 0.5){
+                random_colors.push(colors[i]);
+              }
+          }
+          if (random_colors.length === 0) {
+            random_colors = ['rainbowunicorn'];
+          }
+          return random_colors;
+        }
+        console.log(getRandomInt());
```

Comment:
Having these logs isn't really too helpful without some other text (like console.log("made new cat of color " + color). I imagine they were for debugging but you should remove that when submitting.